### PR TITLE
✨ Add options & request to the generateFileName function

### PR DESCRIPTION
**Pull request by stefanprokopdev**

### CODE REVIEW

File: src/lib/fileUploader/gcpBucket.ts [hunk 1]
```diff
@@ -1,28 +1,29 @@
 import { Bucket } from '@google-cloud/storage';
-import { RequestHandler } from 'express';
+import { Request, RequestHandler } from 'express';
 import * as path from 'path';
 import { Readable } from 'stream';
 import generateFileStamp from '../util/generateFileStamp';
 
-interface Options {
+export interface GcpOptions {
```
Comment (smoliji) [inline src/lib/fileUploader/gcpBucket.ts]: This is unnecessary change for this file.

File: src/lib/fileUploader/gcpBucket.ts [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
     prefix?: string;
     bucket: string;
     credentials: object | string;
     projectId: string;
     public?: boolean;
-    generateFileName?: (params: { originalFile: File; id: string }) => string;
+    generateFileName?: (params: { originalFile: File; id: string }, options?: GcpOptions) => string;
+    request?: Request;
```
Comment (smoliji) [inline src/lib/fileUploader/gcpBucket.ts]: These are options only to initialize the upload, request has no business here.

It should be passed as another argument for `saveFiles` here https://github.com/AckeeCZ/unicore/pull/9/files#diff-f4fcae57a7b2900a50ff24afc03ccd25R85, and then passed as a second argument to `generateFileName` - this way we dont change public api.

**Review by smoliji — CHANGES_REQUESTED** [review]

File: src/lib/fileUploader/gcpBucket.ts (hunk 2 above)
Comment (stefanprokopdev) [reply src/lib/fileUploader/gcpBucket.ts]: :heavy_check_mark: 

`request` added as optional argument for `saveFiles` and `generateFileName` functions

**Comment by smoliji** [comment]
much cleaner now :ok_hand:
